### DISCUSSION TITLE
Modify REST server to change status code returned

### DISCUSSION THREAD

**Discussion by PrasoonPratham** [movementlabsxyz/movement#216] [General]
Can we modify the REST server to change the status code returned by https://aptos.testnet.suzuka.movementlabs.xyz from 404 to 200? Returning 404 prevents the network URL https://aptos.testnet.suzuka.movementlabs.xyz from being added to the Petra wallet. Adding https://aptos.testnet.suzuka.movementlabs.xyz/v1 causes an error when making transactions because Petra appends /v1/xxxx to the URL.

![IMG_2876](https://github.com/user-attachments/assets/a7ae7e5a-9f1c-473c-be62-7004c8b4b8b5)

![IMG_2877](https://github.com/user-attachments/assets/db4ae1de-ed11-4a97-9284-9c5d3021ed04)

**Comment by l-monninger** [comment]
Very likely we are missing one of the services, i.e., some kind of metadata service that Aptos now serves alongside the full-node API. I will look into this and write up an issue when I have determined what needs to be added.

**Comment by l-monninger** [comment]
We are simply missing the `root_handler` attachment as Aptos uses here for the Full Node API: https://github.com/movementlabsxyz/aptos-core/blob/b54d443967ff4ce621a84f7c09f9a59a0b03e01c/api/src/runtime.rs#L261

**Comment by bhenhsi** [comment]
Please follow #236 for updates.

**Reply by robojosef** [reply]
#236 is closed but I still can't connect Petra to Movement Testnet. Is it expected to be working now?
